interpreter/src/symbol_table/var_data/tests.rs

```diff
@@ -103,64 +103,89 @@ mod query {
         }
+        //The variable root is always included in the map. (TODO this might be a bug)
+        m_vars.insert(0, var_m("foo", &[]));
 
         assert_eq!(
-            table.query(&var_m_nullable("foo", &[""]), Direction::Forward),
-            format!("{}", m_vars.first().unwrap())
+            table.query(m_vars.first().unwrap(), Direction::Backward),
+            None
         );
-        for [current, expected] in m_vars.array_windows() {
-            assert_eq!(
-                table.query(current, Direction::Forward),
-                format!("{expected}")
-            );
+        for [a, b] in m_vars.array_windows() {
+            assert_eq!(table.query(a, Direction::Forward).as_ref(), Some(b));
+            assert_eq!(table.query(b, Direction::Backward).as_ref(), Some(a));
         }
-        assert_eq!(table.query(m_vars.last().unwrap(), Direction::Forward), "");
 
-        //Backwards
         assert_eq!(
-            table.query(&var_m_nullable("foo", &[""]), Direction::Backward),
-            format!("{}", m_vars.last().unwrap())
+            table.query(m_vars.last().unwrap(), Direction::Forward),
+            None
         );
-        for [expected, current] in m_vars.array_windows().rev() {
-            assert_eq!(
-                table.query(current, Direction::Backward),
-                format!("{expected}")
-            );
+    }
+
+    #[test]
+    fn nullable_subscripts_can_be_used() {
+        let keys: [&[&str]; 4] = [&["-1"], &["0", "-1"], &["0", "1"], &["a"]];
+        let m_vars: Vec<_> = keys.map(|x| var_m("foo", x)).to_vec();
+
+        let mut table = Table::new();
+        for var in &m_vars {
+            table.set(var, &Value::try_from("Value").unwrap()).unwrap();
         }
+        let boundary = var_m_nullable("foo", &["0", ""]);
         assert_eq!(
-            table.query(m_vars.first().unwrap(), Direction::Backward),
-            ""
+            table.query(&boundary, Direction::Forward).as_ref(),
+            m_vars.get(1)
+        );
+        assert_eq!(
+            table.query(&boundary, Direction::Backward).as_ref(),
+            m_vars.get(2)
         );
     }
 
     #[test]
     fn value_with_no_subscripts() {
         let mut table = super::Table::new();
         let _ = table.set(&var_m("foo", &[]), &Value::try_from("Value").unwrap());
+
         assert_eq!(
             table.query(&var_m_nullable("foo", &[""]), Direction::Forward),
-            ""
+            None
         );
         assert_eq!(
             table.query(&var_m_nullable("foo", &["bar"]), Direction::Forward),
-            ""
+            None
         );
         assert_eq!(
             table.query(&var_m_nullable("foo", &[""]), Direction::Backward),
-            ""
+            None
         );
         assert_eq!(
             table.query(&var_m_nullable("foo", &["bar"]), Direction::Backward),
-            ""
+            None
         );
     }
 
+    ///TODO Potential Bug
```

Comment:
Specify that this might be a C bug and clarity why I think that might be the case.